cmp/grid/Grid.js

```diff
@@ -143,6 +143,16 @@ class GridLocalModel extends HoistModel {
         return model.treeMode && model.store.allRootCount !== model.store.allCount;
     }
 
+    @computed
+    get depthOfLowestLeaf() {
+        let maxDepth = 0;
+        if (this.isHierarchical) {
+            const {records} = this.model.store;
+            records.forEach(record => maxDepth = Math.max(record.treePath.length - 1, maxDepth));
+        }
+        return maxDepth;
+    }
```

Comment:
@lbwexler - this seemed like a potentially expensive reaction to be running against `records` here, for this purpose alone. 

It's possible there is some completely alternate approach that doesn't depend on getting this stat out of the store, but I have not come up with any such alternative.  If we do need this state, could it be calculated/produced more efficiently/centrally within the Store or RecordSet classes?

Reply:
not as worried about the performance of this -- we do a more signifigant O(n) operation when the records change just to generate any transaction -- this seems like a minor addition to that work.  

Not obvious that it would be signifigantly faster computing in recordset itself -- maybe marginally so, in that it has an internal representation of children only that we could limit the check to . could add a getMaxDepth() method on recordset if we wanted to, for clarity and to take advantage of